### PR TITLE
TDD

### PR DESCRIPTION
**Pull request by abel-tefera**
### Summary

- Created tests cases before creating functionality
- Put all the test cases inside `solver_spec.rb` file
- Created a `solver` class with methods `factorial`, `reverse` and `fizzbuzz`

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    end
+
+    it('factorial of 0 should equal 1') do
+      expect(fact_obj.find(0)).to eq(1)
+    end
+
+    it('factorial of a negative number should return an error') do
+      expect { fact_obj.find(-1) }.to raise_error(ArgumentError, 'Argument must be greater than or equal to 0.')
+    end
+  end
```
Comment (julie-ify) [inline spec/solver_spec.rb:38]: - [ ] All your factorial tests are failing. Kindly look into that.

  image

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @abel-tefera and @Patfarmurs,

Good job so far!

## Highlights!
- Fizzbuzz tests ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 1 above)
Comment (DeliceLydia) [reply spec/solver_spec.rb:38]: Addressed

**Review by DeliceLydia — APPROVED** [review]
## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- All tests are passing


### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**